app/src/main/java/com/beemdevelopment/aegis/vault/VaultBackupManager.java

```diff
@@ -64,6 +66,46 @@ public void scheduleBackup(File tempFile, Uri dirUri, int versionsToKeep) {
         });
     }
 
+    private void createBackup(File tempFile)
+            throws VaultRepositoryException, VaultBackupPermissionException {
+        BackupsVersioningStrategy strategy = _prefs.getBackupVersioningStrategy();
+        Uri uri = _prefs.getBackupsLocation();
+        if (uri == null) {
+            throw new VaultRepositoryException("getBackupsLocation returned null");
+        }
+        if (strategy == BackupsVersioningStrategy.SINGLE_BACKUP) {
+            createBackup(tempFile, uri);
+        } else if (strategy == BackupsVersioningStrategy.MULTIPLE_BACKUPS) {
+            int versionsToKeep = _prefs.getBackupsVersionCount();
+            createBackup(tempFile, uri, versionsToKeep);
+        }
```

Comment:
Throw a ``RuntimeException`` if the strategy is anything other than SINGLE_BACKUP or MULTIPLE_BACKUPS.